pkg/app/app.go

```diff
@@ -536,6 +536,10 @@ func (h HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		log.Println("Error: ", err)
 		var e *apperr.AppError
 		if errors.As(err, &e) {
+			if e.Error() == accounts.MissingUsername {
+				http.Redirect(w, r, "/username", http.StatusSeeOther)
```

Comment:
I am not sure about what de-factor is, but it looks like 302(StatusFound) is usually used for login redirect